libs/ui/src/components/MediaItem/MediaItem.vue

```diff
@@ -11,138 +11,137 @@
-    },
-    isLewd: {
-      type: Boolean,
-      default: false,
-    },
-    isDetail: {
-      type: Boolean,
-      default: false,
-    },
-    placeholder: {
-      type: String,
-      default: '',
-    },
-  },
-  data() {
-    return {
-      defaultMimeType: 'image',
-      components: {
-        ImageMedia: defineAsyncComponent(() => import('./type/ImageMedia.vue')),
-        VideoMedia: defineAsyncComponent(() => import('./type/VideoMedia.vue')),
-        AudioMedia: defineAsyncComponent(() => import('./type/AudioMedia.vue')),
-        ModelMedia: defineAsyncComponent(() => import('./type/ModelMedia.vue')),
-        JsonMedia: defineAsyncComponent(() => import('./type/JsonMedia.vue')),
-        IFrameMedia: defineAsyncComponent(
-          () => import('./type/IFrameMedia.vue')
-        ),
-        ObjectMedia: defineAsyncComponent(
-          () => import('./type/ObjectMedia.vue')
-        ),
-        Media: defineAsyncComponent(() => import('./type/UnknownMedia.vue')),
-      },
-      isLewdBlurredLayer: this.isLewd,
-    }
-  },
-  computed: {
-    resolveComponent() {
-      const type = this.mimeType || this.defaultMimeType
-      return this.components[resolveMedia(type) + SUFFIX]
-    },
-    properSrc() {
-      return this.src || this.placeholder
-    },
-  },
-  watch: {
-    animationSrc() {
-      this.updateComponent()
-    },
-  },
-  mounted() {
-    this.updateComponent()
-  },
-  methods: {
-    async updateComponent() {
-      if (this.animationSrc && !this.mimeType) {
-        this.defaultMimeType = await getMimeType(this.animationSrc)
-      }
-    },
-    showContent() {
-      this.isLewdBlurredLayer = false
-    },
-  },
+const resolveComponent = computed(() => {
+  const type = props.mimeType || mimeType.value
+  return components[resolveMedia(type) + SUFFIX]
+})
+const properSrc = computed(() => props.src || props.placeholder)
+
+const updateComponent = async () => {
+  if (props.animationSrc && !props.mimeType) {
+    mimeType.value = await getMimeType(props.animationSrc)
+  }
+}
+
+watch(
+  () => props.animationSrc,
+  () => updateComponent(),
+  {
+    immediate: true,
+  }
+)
+
+const toggleContent = () => {
+  isLewdBlurredLayer.value = !isLewdBlurredLayer.value
 }
+
+defineExpose({ isLewdBlurredLayer })
 </script>
 
 <style lang="scss" scoped>
-.nsfw-blur {
-  backdrop-filter: blur(60px);
-  position: absolute;
-  top: 0;
-  height: 100%;
-  width: 100%;
-  color: #fff;
-  text-transform: capitalize;
+@import '@/styles/abstracts/variables';
+.media-object {
+  .nsfw-blur {
+    backdrop-filter: blur(60px);
+    position: absolute;
+    top: 0;
+    height: 100%;
+    width: 100%;
+    @include ktheme() {
+      color: theme('text-color');
+    }
 
-  .heading-nsfw {
-    font-weight: 700;
-  }
-  .nsfw-desc {
-    max-width: 18.75rem;
-    text-align: center;
+    .nsfw-desc {
+      max-width: 18.75rem;
+    }
   }
-
-  .nsfw-content {
-    cursor: pointer;
-    bottom: 3.125rem;
+  .nsfw-action {
```

Comment:
better to rename this class to `no-border`

check if global.scss have this kind of utility classes already
if it doesn't , move it there
and if it does, use it